### PR TITLE
Task/fix system status

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Fix system status lambda to us status information in calculating whether the website is up/down/degraded

https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/1490

### CODE REVIEW

File: system_status/system_status.py [hunk 1]
```diff
@@ -4,93 +4,135 @@
...
-            'sms': sms_status
+        "body": {
+            "api": api_status,
+            "admin": admin_status,
+            "email": email_status,
+            "sms": sms_status,
         },
-        'statusCode': 200
+        "statusCode": 200,
     }
```
Comment (whabanks) [inline system_status/system_status.py:138]: Let's make sure we know when data fails to be uploaded to the S3 bucket.

```suggestion
    try :
        s3_client.put_object(
            Body=json.dumps(status_data), Bucket=bucket_name, Key=object_key
        )
    except botocore.exceptions.ClientError as e:
        logging.error(
            "System-status lambda error (pushing to s3): {}".format(e)
        )
    else:
        logging.info(
            "Pushed status data to s3: {}".format(
                {
                    "api": api_status,
                    "admin": admin_status,
                    "email": email_status,
                    "sms": sms_status,
                }
            )
        )
    finally:
        return {
            "body": {
                "api": api_status,
                "admin": admin_status,
                "email": email_status,
                "sms": sms_status,
            },
            "statusCode": 200,
        }
```

File: system_status/system_status.py [hunk 2]
```diff
@@ -4,93 +4,135 @@
...
+from database_queries import (
+    low_template_result_group1,
+    low_template_result_group2,
+    medium_template_result_group1,
+    medium_template_result_group2,
+    high_template_result,
+)
+from determine_db_status import determine_status
+from determine_web_status import determine_site_status
+
```
Comment (whabanks) [inline system_status/system_status.py:16]:
```suggestion
import botocore

```

**Review by whabanks — CHANGES_REQUESTED** [review]
Looking good! I tested out some of the queries in blazer. Just a couple of comments / suggestions and we should be good to on my end.

File: system_status/database_queries.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        "high": "276da251-3103-49f3-9054-cbf6b5d74411",
+    },
+    "sms": {
+        "low": "ab3a603b-d602-46ea-8c83-e05cb280b950",
+        "medium": "a48b54ce-40f6-4e4a-abe8-1e2fa389455b",
+        "high": "4969a9e9-ddfd-476e-8b93-6231e6f1be4a",
+    },
+}
+
+TIMING = {
```
[the remark this answers is not in the document]
Comment (jzbahrai) [reply system_status/database_queries.py:19]: I will add a Readme

File: system_status/system_status.py (hunk 1 above)
Comment (jzbahrai) [reply system_status/system_status.py]: fixed!

File: system_status/system_status.py (hunk 2 above)
Comment (jzbahrai) [reply system_status/system_status.py:17]: done

**Review by whabanks — APPROVED** [review]
LGTM!

**Review by andrewleith — COMMENTED** [review]
I see how this checks against notification_status, but where in the code does it verify how fast particular lanes are sending? (i.e. normal should send within 45m)

**Comment by jzbahrai** [comment]
@andrewleith it is based on the timestamp. so I am checked the notifcations within a certain time frame - so if the message that was created 45 minutes ago, now is in "delivered" status - we are good.

**Pull request merged**